app/jobs/fetch_wos_content_job.rb

```diff
@@ -42,9 +42,22 @@ def fetch_wos_content(publication)
     WebOfScience::FetchWosContent.fetch_from_api(publication)
   end
 
+  ##
+  # Find and update or create a new AuthorPublication.
+  # The unique key is the email address and publication.
+  #
+  # Extended for claiming publication without user login with:
+  # - create a new user record with email fetch by WoS API
+  # - create a hash_id based on author email and assign it to AuthorPublication
+  # @param Array<String> emails - an array of found emails
+  # @param <Publication> publication
   def create_or_update_publication_emails(emails, publication)
```

Comment:
Since this functionality lives in the Publication model, instead of doing the same work here you can just call the method for the publication;

`publication.add_author_emails(emails)`